cljfmt/src/cljfmt/core.cljc

```diff
@@ -280,10 +280,22 @@
     (or (symbol-matches-key? (fully-qualified-symbol possible-sym context) key)
         (symbol-matches-key? (remove-namespace possible-sym) key))))
 
+(defn- parent-is-thread-first? [zloc context]
+  (some #(form-matches-key? zloc % context) #{'-> 'as-> 'some-> 'cond->}))
+
+(defn- is-first-parameter? [zloc]
+  (= (z/right (z/leftmost zloc)) zloc))
+
+(defn- adjust-idx [zloc idx context]
+  (cond-> idx
+    (and (parent-is-thread-first? zloc context) (not (is-first-parameter? zloc)))
+    (some-> dec (max 0))))
```

Comment:
Instead of `adjust-idx` which doesn't convey *why* the index is being adjusted, perhaps instead we just split out the predicate:

```clojure
(defn- in-thread-macro? [zloc context]
  (and (form-matches-thread-macro? zloc) (not (first-argument? zloc))))
```

Then we could write:

```clojure
(cond-> idx (in-thread-macro? zloc context) dec)
```

I'm unsure why you have `(some-> dec (max 0))` instead of just `dec`. Under what circumstances would the index be less than or equal to zero?

Reply:
Wouldn't `idx` be zero in this case, if I configured `{cond-> [[:block 0]]}`?
There is a check for `(nil? idx)` in `inner-indent` already. Isn't it nil when there is no indentation rule, like `(-> x foo bar)`?

Reply:
The index is the position of the child in the parent form, IIRC. So, come to thing about it, we could change `(not (first-argument? zloc))` to `(> idx 1)`.

Reply:
Doesn't the index come from the :indents as specified [here](https://github.com/weavejester/cljfmt/blob/master/docs/INDENTS.md)?
So for `:block` it uses default indenting until we reach the index. So if we are in a thread, we should basically hit that index one spot before (ie. decrease the index). If the index is 0, we should keep it at 0.
For `:inner` when index is set, it is restricted to that particular index. So that should also occur one spot earlier (ie. decrease the index). If the index is 0, then we should avoid using this indentation rule, since the spot it was meant for is not present.

I also oversaw the `depth` argument of inner. Here we should check whether the parent of the `top` zloc is wrapped in a thread. I'll commit a fix.

Hope it makes sense.

Reply:
Ah, yes, I believe you're right. I need to refresh myself on the indentation logic as it's been a while since I've looked at it.